packages/web-app-files/src/components/AppBar/ViewOptions.vue

```diff
@@ -1,17 +1,19 @@
 <template>
   <div class="oc-flex oc-flex-middle">
-    <div data-testid="viewmode-switch-buttons" class="oc-button-group oc-visible@s oc-mr-s">
-      <oc-button
-        :appearance="viewModeCurrent === ViewModeConstants.condensedTable ? 'filled' : 'outline'"
-        @click="setViewMode(ViewModeConstants.condensedTable)"
-      >
-        <oc-icon name="menu-line-condensed" fill-type="none" size="small" />
-      </oc-button>
+    <div
+      v-if="viewModes.length"
+      data-testid="viewmode-switch-buttons"
+      class="oc-button-group oc-visible@s oc-mr-s"
+    >
       <oc-button
-        :appearance="viewModeCurrent === ViewModeConstants.default ? 'filled' : 'outline'"
-        @click="setViewMode(ViewModeConstants.default)"
+        v-for="viewMode in viewModes"
+        :key="viewMode.name"
+        v-oc-tooltip="viewMode.label"
+        :appearance="viewModeCurrent === viewMode.name ? 'filled' : 'outline'"
+        :label="viewMode.label"
```

Comment:
I think you need to pass this to $gettext again to make translations actually work